dockstore-webservice/src/main/java/io/dockstore/webservice/core/SourceFile.java

```diff
@@ -263,6 +263,23 @@ public Timestamp getDbUpdateDate() {
         return dbUpdateDate;
     }
 
+    /**
+     * Determine if the specified path matches the sourcefile path.
+     * The paths will match if, after adding a leading slash when it
+     * is missing, the resulting strings are equal, case insensitive.
+     * Thus, an absolute path and an absolute path missing the leading
+     * slash will match, as will paths that are the same string,
+     * whether absolute or relative.
+     */
+    @JsonIgnore
+    public boolean isSamePath(String otherPath) {
+        return otherPath != null && toAbsolutePath(getPath()).equalsIgnoreCase(toAbsolutePath(otherPath));
+    }
+
+    private static String toAbsolutePath(String path) {
```

Comment:
I feel like this is a bit of a misleading name, this doesn't covert a relative path to an absolute path so much as it sticks a "/" in front of a path and assumes it is absolute. Maybe `assumeAbsolutePath` ?